languages/elixir/exercises/concept/enum/.docs/introduction.md

```diff
@@ -0,0 +1,60 @@
+`Enum` is a very useful module that provides a set of algorithms for working with enumerables. It offers sorting, filtering, grouping, counting, searching, finding min/max values, and much more.
+
+In Elixir, an enumerable is any data type that implements the `Enumerable` protocol. The most common of those are lists and maps.
+
+```elixir
+Enum.sort([4, 2, 3, 1])
+# => [1, 2, 3, 4]
+```
+
+```elixir
+Enum.empty?(%{})
+# => true
+```
+
+Many `Enum` functions accept a function as a second argument.
+
+```elixir
+Enum.filter([1, 2, 3, 4, 5], fn x -> x > 3 end)
+# => [4, 5]
+```
+
+The most common `Enum` functions are `map` and `reduce`.
+
+`Enum.map/2` allows you to replace every element in an enumerable with another element. The second argument to `Enum.map/2` is a function that accepts the original element and returns its replacement.
+
+```elixir
+Enum.map([1, 2, 3, 4, 5], fn x ->
+  x * x
+end)
+# => [1, 4, 9, 16, 25]
+```
```

Comment:
This does not introduce any new syntax, does it? If that is correct, this should be removed from the `introduction.md` file (it can be in the `after.md` document though).

Reply:
Technically speaking, none of the code blocks in this whole file introduce any new syntax. The whole concept is about showing off a few important functions, no new syntax.

I feel like this rule is too harsh... is it really in the best interest of a student to learn about new functions and have to rely entirely on a written description of what it does, without being shown any examples?

Reply:
I've thought about this a bit. I think it's worth us coming back to _why_ that rule is in place and what we're trying to achieve with it. Two things:
1) Keep the introductions manageable so they don't require huge reading
2) Add learning through self-discovery not teaching.

(1) could be argued either way. For me, the code is clearer than the language because of how my brain works, so I do think it's adding value.

However, for (2), by spelling out the code in such depth, in the actual example, we're effectively going to be asking the students to copy/paste bits from the introduction into their solution. Effectively we're giving them the lego bricks and asking them to make something. In doing so, I think we lose value. Without the examples, the student has to make sense of the words and think through what they mean (what's an enumberable, what does `Enum.map/2` mean etc) then, using the skills they've picked up from the prereqs, convert those words into code. That puts the brain through a lot more work but it means people will learn loads more through the process. 

So I think the code examples should go into the hints file, so if a student is stuck, or doesn't want to go through that learning, they can still see tangible examples, but it's not given to them for free if they want to work it out themselves (which is what I personally would want as a student).

This isn't something I'm saying I think there is an absolute right/wrong for. I think it's more how we want Exercism to be teaching things. And I might be totally incorrect about what's best. But that's where my thinking is currently.

Reply:
I've created a PR to add a note about using code samples in `hints.md` files (which we currently did not allow): https://github.com/exercism/v3/pull/1602

Reply:
Thank you for the detailed explanation. It definitely helped.

@ErikSchierboom I left a single code example in the introduction. It uses a function that is not used in the exercise at all so nothing to copy-paste. It's supposed to be a reminder that anonymous functions can be created and passed as an argument at the same time. It was technically already mentioned in the anonymous function concept after.md, but I think students coming from languages without first-class functions might struggle with that idea. Is that an acceptable compromise?